.github/workflows/python-package.yml

```
@@ -50,3 +53,38 @@ jobs:
     - name: Generate coverage report
       run: |
         poetry run coverage report
+
+  build-and-publish:
+    runs-on: ubuntu-latest
+    needs: tests
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.11"
+
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+      with:
+        # Version of Poetry to use
+        version: 1.4.1
+
+    - name: Build distribution tarball and wheel
+      # TODO: restrict this to master branch only
+      run: |
+        poetry build
+
+    - name: Publish distribution 📦 to Test PyPI
+      # publish to testpypi on all commit just for testing this PR
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.TEST_PYPI_API_TOKEN }}
+        repository-url: https://test.pypi.org/legacy/
+
+    - name: Publish distribution 📦 to PyPI
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.PYPI_API_TOKEN }}
```

Poetry can also do the publish for you. I haven't tested publishing to test pypi before, but according to the configs I found I think I have it correct.

```suggestion
    - name: Install Dependencies
      run: |
        poetry install
        
    - name: Publish distribution 📦 to Test PyPI
      # publish to testpypi on all commit just for testing this PR
      run |
        # I believe publishing with password is depreciated and you you need a token instead?
        poetry config pypi-token.test-pypi ${{ secrets.TEST_PYPI_API_TOKEN }} 
        poetry config repositories.test-pypi https://test.pypi.org/legacy/
        poetry publish -r test-pypi --build

    - name: Publish distribution 📦 to PyPI
      if: startsWith(github.ref, 'refs/tags')
      run |
      poetry config pypi-token.pypi ${{ secrets.PYPI_API_TOKEN }}
      poetry publish --build
```